File: 118-pascals-triangle/118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    const triangle = [];
    for (let i=0; i<numRows; i++){
        triangle.push([]);
    }
    for (let i=0; i<triangle.length; i++){
        if (i===0){
            triangle[0].push(1);
        } else {
            for(let j=0; j<=i; j++){
                if((!triangle[i-1][j-1]) || (!triangle[i-1][j])) {
                    triangle[i].push(1);
                } else {
                    triangle[i].push(triangle[i-1][j-1] + triangle[i-1][j]);
                }
            }
        }
    }
    return triangle;
};

File: 118-pascals-triangle/118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = []
        for i in range(numRows):
            triangle.append([])
        for i in range(len(triangle)):
             if i == 0:
                    triangle[i].append(1)
             else:
                for j in range(i+1):
                        if j-1 < 0 or j >= len(triangle[i-1]):
                            triangle[i].append(1)
                        else:
                            triangle[i].append(triangle[i-1][j-1] + triangle[i-1][j])
        return triangle